jupyterlab/commands.py

```diff
@@ -564,8 +574,7 @@ def __init__(self, options):
         self.core_data = options.core_config._data
         self.info = self._get_app_info()
         self.kill_event = options.kill_event
-        # TODO: Make this configurable
-        self.registry = 'https://registry.npmjs.org'
+        self.registry = options.registry
```

Comment:
@vidartf with the latest commit, `registry` is moved to `AppOptions`. But the side effect is that the default registry will be the yarn registry not the npm.

Reply:
I thought that was intended? Does it not support the full registry API?

Reply:
It seems so. But as they plan to deprecate yarn registry, I'm wondering if this is a good idea for the long term?

Reply:
If it is kept separate, and the yarn registry is deprecated, won't it still break equally hard? What will be the fix if/when the yarn registry is deprecated?

Reply:
From my understanding the deprecation will occur in parallel to the 2.0 release. As it remove the registry from the package url, this fix won't do anything. But if the yarn default registry site is down it will break the extension manager requests. On the other hand, as jupyterlab ships with its own yarn. Moving to 2.0 will be under control and when doing so, the default registry should probably be set back to npm and not yarn. What could I do to keep track of this?

Reply:
I know there has been discussion on this before, but I cannot find it again: why not use the npm registry as the default?

Reply:
Answering my own question: because the yarn registry is the default of jlpm, and the one used in our lockfile in staging.